### DISCUSSION TITLE
Masking Sensitive information in TaskInstance Details

### DISCUSSION THREAD

**Discussion by Shadowsong27** [apache/airflow#25081] [Q&A]
Hi folks!

I have tried masking the sensitve information by following the docs [here](https://airflow.apache.org/docs/apache-airflow/stable/security/secrets/mask-sensitive-values.html), those information are correctly masked in Variables, logs, and connections as expected, but apparently the arguments passed into the Operator will still show up under the `Task Instance Details` tab. 

Is this an expected behaviour? How we could mask those too?

Thanks in advance and apologies if similar question has been asked somewhere else.

**Comment by potiuk** [comment]
You cannot. You should use connections and secret variables to pass sensittive data to your tasks. you should not pass them via arguments because this is impossible or very difficult to "hide" - all those parameters are passed verbatim in many ways (depedning on the deployment they might be part of K8S POD details or celery worker command passed via messaging queue in celery. 

Secret variables and Connections were specifically designed to provide a way of passing secret information that can be hidden from view. You also have all kinds of Secrets Manager integration in airflow - even if you want to dynamically pass a secret between  tasks you can simply store such a secret infromation in your Secrets manager with dynamic nam and pass only "the dynamic variable name" trough XCom to other tasks. There are various ways you can do it - you can even interact directly with secret managers via hooks if you want.
